Library/Homebrew/utils/bottles.rb

```diff
@@ -0,0 +1,104 @@
+require "tab"
+require "extend/ARGV"
+
+module Utils
+  class Bottles
+    class << self
+      def tag
+        @bottle_tag ||= "#{ENV["HOMEBREW_SYSTEM"]}-#{ENV["HOMEBREW_PROCESSOR"]}".downcase.to_sym
+      end
+
+      def built_as?(f)
+        return false unless f.installed?
+        tab = Tab.for_keg(f.installed_prefix)
+        tab.built_as_bottle
+      end
+
+      def file_outdated?(f, file)
+        filename = file.basename.to_s
+        return unless f.bottle && filename.match(Pathname::BOTTLE_EXTNAME_RX)
+
+        bottle_ext = filename[bottle_native_regex, 1]
+        bottle_url_ext = f.bottle.url[bottle_native_regex, 1]
+
+        bottle_ext && bottle_url_ext && bottle_ext != bottle_url_ext
+      end
+
+      def native_regex
+        /(\.#{tag}\.bottle\.(\d+\.)?tar\.gz)$/o
```

Comment:
That will always be a point of confusion for me, as different languages tend to abbreviate this differently. I guess I was most influenced by the recent demand for replacing `/PATTERN/` with `/REGEX/` (note the missing “P”) in the output of `brew help`. :wink: